service/controller/clusterapi/v17/resources/tenantclients/create.go

```diff
@@ -51,12 +51,18 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 	{
 		g8sClient, err = r.tenant.NewG8sClient(ctx, key.ClusterID(&cluster), key.ClusterAPIEndpoint(cluster))
 		if err != nil {
-			return microerror.Mask(err)
+			r.logger.LogCtx(ctx, "level", "warning", "message", "failed to create tenant cluster g8s client", "stack", microerror.Stack(microerror.Mask(err)))
```

Comment:
We should have more specific error matching here towards the errors we expect. Then we should return unexpected errors like we do when we have no idea what is going wrong. This code would now hide unexpected problems.

Reply:
Sorted.